crates/bevy_tasks/src/bevy_executor.rs

```diff
@@ -0,0 +1,1296 @@
+        // LOCAL_QUEUE in any way.
+        f(unsafe { &mut *tls.get() })
+    })
+}
+
+struct LocalQueue {
+    local_queue: VecDeque<Runnable>,
+    local_active: Slab<Waker>,
+}
+
+impl Drop for LocalQueue {
+    fn drop(&mut self) {
+        for waker in self.local_active.drain() {
+            waker.wake();
+        }
+
+        while self.local_queue.pop_front().is_some() {}
+    }
+}
+
+struct ThreadLocalState {
+    executor: AtomicPtr<State>,
+    stealable_queue: ConcurrentQueue<Runnable>,
+    thread_locked_queue: ConcurrentQueue<Runnable>,
+}
+
+impl Default for ThreadLocalState {
+    fn default() -> Self {
+        Self {
+            executor: AtomicPtr::new(core::ptr::null_mut()),
+            stealable_queue: ConcurrentQueue::bounded(512),
+            thread_locked_queue: ConcurrentQueue::unbounded(),
+        }
+    }
+}
+
+/// A task spawner for a specific thread. Must be created by calling [`TaskPool::current_thread_spawner`]
+/// from the target thread.
+///
+/// [`TaskPool::current_thread_spawner`]: crate::TaskPool::current_thread_spawner
+#[derive(Clone, Debug)]
+pub struct ThreadSpawner<'a> {
+    thread_id: ThreadId,
+    target_queue: &'static ConcurrentQueue<Runnable>,
+    state: Arc<State>,
+    _marker: PhantomData<&'a ()>,
+}
+
+impl<'a> ThreadSpawner<'a> {
+    /// Spawns a task onto the specific target thread.
+    pub fn spawn<T: Send + 'static>(
+        &self,
+        future: impl Future<Output = T> + Send + 'static,
+    ) -> Task<T> {
+        // SAFETY: T and `future` are both 'static, so the Task is guaranteed to not outlive it.
+        unsafe { self.spawn_scoped(future) }
+    }
+
+    /// Spawns a task onto the executor.
+    ///
+    /// # Safety
+    /// The caller must ensure that the returned Task does not outlive 'a.
+    pub unsafe fn spawn_scoped<T: Send + 'a>(
+        &self,
+        future: impl Future<Output = T> + Send + 'a,
+    ) -> Task<T> {
+        let mut active = self.state.active();
+
+        // Remove the task from the set of active tasks when the future finishes.
+        let entry = active.vacant_entry();
+        let index = entry.key();
+        let state = self.state.clone();
+        let future = AsyncCallOnDrop::new(future, move || drop(state.active().try_remove(index)));
+
+        // Create the task and register it in the set of active tasks.
+        //
+        // SAFETY:
+        //
+        // - `future` is `Send`. Therefore we do not need to worry about what thread
+        //   the produced `Runnable` is used and dropped from.
+        // - `future` is not `'static`, but the caller must make sure that the Task
+        //   and thus the `Runnable` will not outlive `'a`.
+        // - `self.schedule()` is `Send`, `Sync` and `'static`, as checked below.
+        //   Therefore we do not need to worry about what is done with the
+        //   `Waker`.
+        let (runnable, task) = unsafe {
+            Builder::new()
+                .propagate_panic(true)
```

Comment:
For some reason I thought this was the default behavior, but I looked it up and I was wrong.

This is correct, and I'll have to adjust forte to match.